hibernate-core/src/main/java/org/hibernate/dialect/PostgreSQLDialect.java

```diff
@@ -913,6 +915,14 @@ public String getSelectClauseNullString(int sqlType, TypeConfiguration typeConfi
 		return "cast(null as " + typeConfiguration.getDdlTypeRegistry().getDescriptor( sqlType ).getRawTypeName() + ")";
 	}
 
+	@Override
+	public String getSelectClauseNullString(SqlTypedMapping sqlType, TypeConfiguration typeConfiguration) {
+		final String castTypeName = typeConfiguration.getDdlTypeRegistry()
+				.getDescriptor( sqlType.getJdbcMapping().getJdbcType().getDdlTypeCode() )
+				.getCastTypeName( sqlType.toSize(), (SqlExpressible) sqlType.getJdbcMapping(), typeConfiguration.getDdlTypeRegistry() );
+		return "cast(null as " + castTypeName + ")";
+	}
+
```

Comment:
@beikov could you please add javadoc or at least a comment explaining the particular reasons this override is necessary here

Reply:
Done